code/modules/clothing/shoes/miscellaneous.dm

```diff
@@ -410,6 +410,33 @@
 	item_state = "kneesock"
 	species_fit = list(INSECT_SHAPED)
 
+/obj/item/clothing/shoes/kneesocks/attackby(obj/item/weapon/W, mob/user)
+	..()
+	if(istype(W, /obj/item/weapon/soap))
+		to_chat(user, "<span class='notice'>You place \the [W] into \the [src].</span>")
+		if(do_after(user, src, 1 SECONDS))
+			user.drop_item(src)
+			if(!user.drop_item(W))
+				to_chat(user, "<span class='warning'>You can't let go of \the [W].</span>")
+				return
+			var/obj/item/weapon/soap_sock/I = new (get_turf(user))
+			user.put_in_hands(I)
+			W.transfer_fingerprints_to(I)
+			I.base_soap = W
+			W.forceMove(null)
+			qdel(src)
```

Comment:
```suggestion
			W.forceMove(I)
			src.forceMove(I)
```

You should also assign the kneesocks (`src`) to a variable on the `soap_sock`, akin to `base_soap`, so that you can pop it back out afterwards.

Reply:
Fixed in latest commit